bindings/rust/evmc-vm/src/lib.rs

```diff
@@ -69,14 +103,40 @@ pub struct ExecutionMessage {
 }
 
 /// EVMC transaction context structure.
-pub type ExecutionTxContext = ffi::evmc_tx_context;
+#[derive(Debug, Copy, Clone, Hash, PartialEq)]
+pub struct ExecutionTxContext<'a> {
+    #[doc = "The transaction gas price."]
+    pub tx_gas_price: Bytes32,
+    #[doc = "The transaction origin account."]
+    pub tx_origin: Address,
+    #[doc = "The miner of the block."]
+    pub block_coinbase: Address,
+    #[doc = "The block number."]
+    pub block_number: i64,
+    #[doc = "The block timestamp."]
+    pub block_timestamp: i64,
+    #[doc = "The block gas limit."]
+    pub block_gas_limit: i64,
+    #[doc = "The block previous RANDAO (EIP-4399)."]
+    pub block_prev_randao: Bytes32,
+    #[doc = "The blockchain's ChainID."]
+    pub chain_id: Bytes32,
+    #[doc = "The block base fee per gas (EIP-1559, EIP-3198)."]
+    pub block_base_fee: Bytes32,
+    #[doc = "The blob base fee (EIP-7516)."]
+    pub blob_base_fee: Bytes32,
+    #[doc = "The array of blob hashes (EIP-4844)."]
+    pub blob_hashes: &'a [Bytes32],
+    #[doc = "The array of transaction initcodes (TXCREATE)."]
+    pub initcodes: &'a [evmc_tx_initcode],
+}
```

Comment:
Same thing here. Before `ExecutionTxContext` was just a alias for the C type which has pointers and length for `blob_hashes` and `initcodes`. I created a new type which converts those fields to slices so that I do not have to convert them each time I use them.

Reply:
As discussed offline, this change is somewhat functional: removing boilerplate in the evmrs implementation. Nevertheless, it is not required for the stepN interface. I will encourage you to find place for this translation layer in evmrs instead of evmc. This may not be a trivial change but would dramatically reduce the conflict surface when we need to roll up a new evmc version. 
Unfortunately, it may not be easy to submit this change ot evmc, as it will break compatibility with existing rust evms. I believe that this code should be used by every implementation, though.